src/main/java/de/zalando/aruha/nakadi/service/EventPublisher.java

```diff
@@ -0,0 +1,141 @@
+package de.zalando.aruha.nakadi.service;
+
+import de.zalando.aruha.nakadi.domain.BatchItem;
+import de.zalando.aruha.nakadi.domain.BatchItemResponse;
+import de.zalando.aruha.nakadi.domain.EventPublishResult;
+import de.zalando.aruha.nakadi.domain.EventPublishingStatus;
+import de.zalando.aruha.nakadi.domain.EventPublishingStep;
+import de.zalando.aruha.nakadi.domain.EventType;
+import de.zalando.aruha.nakadi.exceptions.EventPublishingException;
+import de.zalando.aruha.nakadi.exceptions.EventValidationException;
+import de.zalando.aruha.nakadi.exceptions.InternalNakadiException;
+import de.zalando.aruha.nakadi.exceptions.NoSuchEventTypeException;
+import de.zalando.aruha.nakadi.exceptions.PartitioningException;
+import de.zalando.aruha.nakadi.partitioning.PartitionResolver;
+import de.zalando.aruha.nakadi.repository.TopicRepository;
+import de.zalando.aruha.nakadi.repository.db.EventTypeCache;
+import de.zalando.aruha.nakadi.validation.EventTypeValidator;
+import de.zalando.aruha.nakadi.validation.ValidationError;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+
+public class EventPublisher {
+    private static final Logger LOG = LoggerFactory.getLogger(EventPublisher.class);
+
+    private final TopicRepository topicRepository;
+    private final EventTypeCache cache;
+    private final PartitionResolver partitionResolver;
+
+    public EventPublisher(final TopicRepository topicRepository,
+                         final EventTypeCache cache,
```

Comment:
Rename to `eventTypeCache`? From reading a variable just called `cache` in the code, I don't know what is actually cached.